modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -146,6 +154,65 @@
                table-schema->metabase-field-info
                set)})
 
+(def ^:private ^DateTimeFormatter bq-ts-str-formatter
+  "For some crazy reason, the TIMESTAMP String values returned from [[FieldValue]] look like `1577836800.0`, so create
+  a parser for that.  This is basically '<epoch seconds>.<millis>'."
+  (.toFormatter (doto (DateTimeFormatterBuilder.)
+                  (.appendValue ChronoField/INSTANT_SECONDS 1 19 SignStyle/NEVER)
+                  (.appendLiteral \.)
+                  (.appendValue ChronoField/MILLI_OF_SECOND))))
+
+(defn- bq-ts-str->instant [ts-str]
+  (let [parsed (.parse bq-ts-str-formatter ts-str)]
+    (Instant/from parsed)))
+
+(defn- FieldValueList->fingerprint-structure [fields truncation-size ^FieldValueList values]
+  (map-indexed (fn [idx ^FieldValue fv]
+                 (let [{:keys [:database_type]} (nth fields idx)
+                       v (.getValue fv)]
+                      (if (string? v)
+                        (cond (= "TIMESTAMP" database_type)
+                              (bq-ts-str->instant v)
+
+                              (some? truncation-size)
```

Comment:
I'm not sure we need to worry about truncation here. The point of truncation is to not send stuff over the wire and have it in memory. Once that's already done, we might as well fingerprint it. The fingerprinters do not accumulate the original value but stats about the values. So there's no point in truncating the results of the query. It's better to truncate in the query. And if not, don't bother with it.

Reply:
Ah, good point.  Will remove the param and truncation.